ethers-signers/src/trezor/app.rs

```diff
@@ -56,9 +61,41 @@ impl TrezorEthereum {
         Ok(())
     }
 
+    fn get_cached_session() -> Result<Option<Vec<u8>>, TrezorError> {
+        let mut session = [0; 32];
+
+        let path = env::current_dir()
+            .map_err(|e| TrezorError::CacheError(e.to_string()))?
+            .join("cache")
+            .join("trezor.session");
+
+        if let Ok(file) = fs::File::open(path) {
+            let mut file = std::io::BufReader::new(file);
+            file.read(&mut session).map_err(|e| TrezorError::CacheError(e.to_string()))?;
+
+            Ok(Some(session.to_vec()))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn save_session(&mut self, session_id: Vec<u8>) -> Result<(), TrezorError> {
+        let mut path =
+            env::current_dir().map_err(|e| TrezorError::CacheError(e.to_string()))?.join("cache");
+        fs::create_dir_all(&path).map_err(|e| TrezorError::CacheError(e.to_string()))?;
+        path = path.join("trezor.session");
+
+        let file = fs::File::create(path).map_err(|e| TrezorError::CacheError(e.to_string()))?;
+        let mut file = std::io::BufWriter::new(file);
```

Comment:
same, if the session id is only 32byte, probably not worth it, `fs::write` should do just fine I believe